### ISSUE TITLE
log scale ticks for colorbar in heatmap()

### ISSUE THREAD

**Issue by berkeragir** [mwaskom/seaborn#679]
Hello,

I had trouble trying to set ticks for colorbar generated using the heatmap() function. I set the norm to a LogNorm object, which resulted in correct visual distribution for the heatmap. Afterwards, I wanted to set the labels of the colorbar and also set the ticks myself. The default ticker determines ticks based on the original values instead of the log scale. I tried changing the ticks using the axes for the colorbar as follows, but using the set_yticks() method didn't work:
```
    ax = sns.heatmap(data, cmap="YlGnBu", square=True,
                     yticklabels=yticks, xticklabels=xticks,
                     norm=LogNorm(vmin=data.min(), vmax=data.max()))

    cax = plt.gcf().axes[-1]
    cax.set_yticks(new_ticks)
```

This basically did not change the ticks (but gave no error as well). I received this warning though: "Use the colorbar set_ticks() method instead." I could not find a way to get the original colorbar object. 

I found out that in matrix.py, a ticker object is defined in a hard-coded way. Instead of setting the cbar value to False in heatmap() call, I think adding a 'log' boolean variable is quite useful here. The following change to matrix.py worked for me:
```     
    # Possibly add a colorbar
        if self.cbar:
            ticker = mpl.ticker.MaxNLocator(6)

            if 'log' in self.cbar_kws:
                if self.cbar_kws.pop('log'):
                    ticker = mpl.ticker.LogLocator()
```
With this change, I was able to set the ticks as follows:
```
    ax = sns.heatmap(data, cmap="YlGnBu", square=True,
                     yticklabels=yticks, xticklabels=xticks, cbar_kws={'log': True},
                     norm=LogNorm(vmin=data.min(), vmax=data.max()))

    cax = plt.gcf().axes[-1]
    cax.set_yticklabels(labels)
```

Do you think this can be incorporated or a better solution for enabling log scaling easily can be implemented?

Best.

**Comment by berkeragir** [comment]
Hello again,

I didn't notice the discussion here: https://github.com/mwaskom/seaborn/issues/459

I'll look into that and see how to proceed.

**Comment by mwaskom** [comment]
I think that's held up by an incorrect test. Feel free to fork, fix, and do a separate PR.

**Comment by Overdrivr** [comment]
@berkeragir @mwaskom I have the feeling that both discussion were not that much related, I still cannot manage to set a logarithmic scale on the heatmap to display properly. Using logNorm, the colorbar does not display any tick. Any idea ?

I found this SO post that is related: http://stackoverflow.com/questions/36898008/seaborn-heatmap-with-logarithmic-scale-colorbar
